packages/experience-builder-sdk/src/blocks/editor/VisualEditorContext.tsx

```diff
@@ -115,12 +120,13 @@ export function VisualEditorContextProvider({
   // afterward, this effect fetches the respective entities.
   useEffect(() => {
     const resolveEntities = async () => {
-      setEntitiesFetched(false);
       const dataSourceEntityLinks = Object.values(dataSource || {});
-      await entityStore.fetchEntities([
-        ...dataSourceEntityLinks,
-        ...(designComponentsRegistry.values() || []),
-      ]);
+      const entityLinks = [...dataSourceEntityLinks, ...(designComponentsRegistry.values() || [])];
+      const deferredEntities = entityStore.fetchEntities(entityLinks);
+      // Only update the state and rerender when we're actually fetching something
+      if (deferredEntities === false) return;
```

Comment:
fthis ` === false` is unexpected because return values of the `fetchEntities()` is not obvious. I made comment on this on the `fetchEntities()` code below.

And IMO function which returns sync `false` and `Promise<Entry[]>` is very confusing. And even the call pattern is unusual because of that :)

Reply:
You're right.
I changed the response of the function and added an explanatory JSDoc why this is helpful for us to avoid re-renders.